src/main/java/com/amazon/ion/impl/macro/MutableMacroTable.kt

```diff
@@ -0,0 +1,12 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package com.amazon.ion.impl.macro
+
+class MutableMacroTable(private val antecedent: MacroTable) : MacroTable {
```

Comment:
In practice at present there's only ever a single antecedent, the system macro table or the empty. 

You're right though that this could be more of a problem when we have general encoding module support and that the interface seen here allows it to be a problem for an implementer now. 

In the general case a flat mutable table does make it easier to add and shadow values, but it makes it more difficult to hand a consistent view of some values off to a consumer without transcribing the values entirely (I'm thinking of a case like e.g. Trino where we may want to hand off a tuple of <bytes, interpretive context>, not the present state of affairs here).  

Definitely something to chew on. I'd be happy to go with the consensus on this one, whatever it is. @tgregg any thoughts?

As an aside- if you've got a better name for _anything_ here please let me know.

Reply:
I hesitate to proclaim that one alternative is better than the other before we've done some performance analysis, so I'm happy to go with the proposed modeling for now and swap it out if we identify that a flat table is typically better. I, too, suspect that both have benefits for some use cases and drawbacks for others.